core/hotspot/rule.go

```diff
@@ -80,6 +80,10 @@ type Rule struct {
 	// if ParamIndex is great than or equals to zero, ParamIndex means the <ParamIndex>-th parameter
 	// if ParamIndex is the negative, ParamIndex means the reversed <ParamIndex>-th parameter
 	ParamIndex int `json:"paramIndex"`
+	// ParamKey is the key in context attachments map.
+	// ParamKey can be used as a supplement to ParamIndex to facilitate rules to quickly obtain parameter from a large number of parameters
+	// Mutually exclusive with ParamIndex
+	ParamKey string `json:"paramKey"`
```

Comment:
the function `func (r *Rule) Equals(newRule *Rule) bool` need check this field：`ParamKey `

Reply:
Otherwise, Maybe we need add  another unit test for `func (r *Rule) Equals(newRule *Rule) bool`